### PR TITLE
Peters workdir

### PR DESCRIPTION
**Pull request by CapsRePeaT**

### CODE REVIEW

File: src/common/include/common.h [hunk 1]
```diff
@@ -17,6 +27,39 @@ enum class ShapeType {
...
+			: type_(type), id_(id), representation_id_(representation_id) {}
+	[[nodiscard]] bool operator==(const GlobalId& other) const = default;
+	[[nodiscard]] constexpr InstanceId id() const {
+		return id_;
+	}
+	[[nodiscard]] constexpr RepresentationId representation_id() const {
+		return representation_id_;
+	}
+	[[nodiscard]] constexpr InstanceType type()
+			const {
```
Comment (Saiel) [inline src/common/include/common.h:52]: kek

File: src/core/include/databases.h [hunk 2]
```diff
@@ -38,45 +41,59 @@ class HeatmapStorage {
 	using Heatmaps   = std::vector<Heatmap>;
 	HeatmapStorage() = default;
 	HeatmapStorage(std::vector<float> x_steps_mv, std::vector<float> y_steps_mv,
-	               const std::vector<float>& temperature);
-
+	               const std::vector<float>& temperature, Box3D design_borders);
+	[[nodiscard]] Box3D design_borders() const { return design_borders_; }
+	[[nodiscard]] float MinStep() const;
```
Comment (CapsRePeaT) [inline src/core/include/databases.h:46]: forgot to implement

File: src/core/include/databases.h [hunk 3]
```diff
@@ -38,45 +41,59 @@ class HeatmapStorage {
...
+	[[nodiscard]] const Heatmaps& heatmaps() const { return heatmaps_; }
+	[[nodiscard]] const std::vector<float>& x_steps() const { return x_steps_; }
+	[[nodiscard]] const std::vector<float>& y_steps() const { return y_steps_; }
+ private:
 	size_t layers_count_ = 0;  // IST in T2D file
 	std::vector<float> x_steps_;
 	std::vector<float> y_steps_;
+	Box3D design_borders_;
```
Comment (CapsRePeaT) [inline src/core/include/databases.h:54]: representation_borders_, should be changed everywere

File: src/core/include/shapes.h [hunk 4]
```diff
@@ -5,37 +5,45 @@
...
-		return representation_id_;
-	}
 
+
+enum class HilightSetup {
+	none = 0,
+	basic,
+	stack_hilight,
+	main_hilight
+};
```
Comment (CapsRePeaT) [inline src/core/include/shapes.h:15]: add comments here

File: src/core/include/shapes.h [hunk 5]
```diff
@@ -5,37 +5,45 @@
...
 			: id_(id), layer_id_(layer_id), bbox_(std::move(bbox_mv)) {}
-	[[nodiscard]] GlobalShapeId id() const { return id_; }
+	[[nodiscard]] GlobalId id() const { return id_; }
 	[[nodiscard]] const Box3D& bbox() const { return bbox_; }
 	void setBox(Box3D box_mv) { bbox_ = std::move(box_mv); }
-
+	VisualisationOptions& visualisation_options() {
+		return visualisation_options_;
+	}
+	[[nodiscard]] LayerId layer_id() const { return layer_id_; }
```
Comment (CapsRePeaT) [inline src/core/include/shapes.h:40]: add comment to describe diff with global id

File: src/core/include/databases.h [hunk 6]
```diff
@@ -17,16 +18,18 @@ class GeomStorage {
 	void Clear() { shapes_.clear(); }
 
  private:
+	// we assume that shape id equual to plase of shape in the array.
```
Comment (Saiel) [inline src/core/include/databases.h:21]: Bad assuming. Need to be fixed in the future. Possible and obvious solution is to make getters in such classes (`GetSmth`, to determinate complexity of getters), where it finds nedded object in corresponding container by stored id

File: src/core/src/databases.cpp [hunk 7]
```diff
@@ -14,10 +14,29 @@ MetadataPack FileRepresentation::GetShapeMetadata(const ShapeId /*id*/) const {
...
+
 HeatmapStorage::HeatmapStorage(std::vector<float> x_steps_mv,
                                std::vector<float> y_steps_mv,
-                               const std::vector<float>& temperature)
-		: x_steps_(std::move(x_steps_mv)), y_steps_(std::move(y_steps_mv)) {
+                               const std::vector<float>& temperature,
+                               Box3D design_borders)
+		: x_steps_(std::move(x_steps_mv)),
+			y_steps_(std::move(y_steps_mv)), 
+		  design_borders_(std::move(design_borders)) {
```
Comment (Saiel) [inline src/core/src/databases.cpp:39]: suffix `_mv` for design_borders

**Review by Saiel — APPROVED** [review]

**Pull request merged**